src/Moat/Pretty/Swift.hs

```diff
@@ -268,3 +268,30 @@ prettyPrivateTypes indents = go
 onLast :: (a -> a) -> [a] -> [a]
 onLast _ [] = []
 onLast f (x : xs) = x : map f xs
+
+-- | Copy protocols from the parent type to upper bounds of generic type
+--   parameters.
+--
+--   This is needed for protocols with compiler-synthesized implementations
+--   (similar to 'deriving stock'), of which there are currently three:
+--
+--   - 'Equatable'
+--   - 'Hashable'
+--   - 'Codable'
+--
+--   See the [Swift documentation](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/protocols#Adopting-a-Protocol-Using-a-Synthesized-Implementation).
+addTyVarBounds :: [String] -> [Protocol] -> [String]
+addTyVarBounds tyVars protos =
+  let synthesizedProtos = filter isSynthesized protos
+   in case synthesizedProtos of
+        [] -> tyVars
+        _ ->
+          let bounds = ": " ++ intercalate " & " (map prettyProtocol synthesizedProtos)
+           in map (++ bounds) tyVars
+  where
+    isSynthesized :: Protocol -> Bool
+    isSynthesized = \case
+      Hashable -> True
+      Codable -> True
+      Equatable -> True
+      _ -> False
```

Comment:
Could you put this into the top `let` block and change `_` to `OtherProtocol _`? The latter enforces you consider whether any added Protocols are synthesized.